firewood/src/storage/buffer.rs

```diff
@@ -642,6 +642,24 @@ mod tests {
             .join("firewood")
     }
 
+    fn create_state_cache(state_path: PathBuf, disk_requester: DiskBufferRequester) -> Arc<CachedSpace> {
```

Comment:
Every single call to `CachedSpace::new` is wrapped by `Arc::new`, so please make `CachedSpace::new` return an `Arc`. For these one-offs that do not use the configuration, perhaps `CachedSpace::new_small()` which only takes the arguments listed here makes sense.

Reply:
Dear Ron, if I understand correctly, this will make the scope of this PR a bit bigger, _love it_.

I've checked every call to `CachedSpace::new` and you're right, it's always wrapped by `Arc::new`.

>For these one-offs that do not use the configuration
I couldn't find any instance of `CachedSpace::new` that didn't receive a `cfg: &StoreConfig` parameter, perhaps I don't understand what you mean on that one.

For now I'll go ahead and try to make `CachedSpace::new` return an `Arc` wrapped result on a separate commit that will touch the following files:
 - `firewood/src/db.rs`
 - `firewood/src/storage/buffer.rs`

Reply:
[done deal](https://github.com/ava-labs/firewood/pull/287/commits/c97d979a1774a9c98bcb1b3cc6630676e390b267) with the first request Ron.

Reply:
@richardpringle 's comment about .into() is much better!